test/keys.c

```diff
@@ -0,0 +1,131 @@
+        "-----END CERTIFICATE-----\r\n",
+        "-----BEGIN EC PRIVATE KEY-----\r\n"
+        "Proc-Type: 4,ENCRYPTED\r\n"
+        "DEK-Info: AES-256-CBC,D0E58BC6B403768675587712ED20447E\r\n"
+        "\r\n"
+        "6hkZev34BDzdOo3qaTQpTsFeQc86zXBV99m0LYpojXmYx81mwXBzNTweV+XbwK9v\r\n"
+        "289SJGs01Gq0H9uL79edR1uwEKSye0OMKiR5Gy46DFXYneBMVVrK/6TgQwPq35W0\r\n"
+        "z0wO+AYpRbdvXUkJJE4sK+tuVeCPpfg6z5hAnq35hcI=\r\n"
+        "-----END EC PRIVATE KEY-----\r\n",
+        "bob"
+    },
+    {
+        { (const uint8_t*)"eve", sizeof("eve") - 1 },
+        "-----BEGIN CERTIFICATE-----\r\n"
+        "MIIBYjCCAQcCCQCzbzjgbS2cBjAKBggqhkjOPQQDAjBIMQswCQYDVQQGEwJVUzEV\r\n"
+        "MBMGA1UEBwwMRGVmYXVsdCBDaXR5MQwwCgYDVQQKDANEUFMxFDASBgNVBAMMC0RQ\r\n"
+        "UyBUZXN0IENBMB4XDTE4MDEzMTE4MDgzNloXDTI4MDEyOTE4MDgzNlowKTELMAkG\r\n"
+        "A1UEBhMCVVMxDDAKBgNVBAoMA0RQUzEMMAoGA1UEAwwDZXZlMFkwEwYHKoZIzj0C\r\n"
+        "AQYIKoZIzj0DAQcDQgAEL/rftnvMRbadGtB836rPqBWBIZ1Jj2CDj5tVbxrqE+XS\r\n"
+        "8MJVvEMXuepE6IqvbAdtemHHZbEo/XA1YtOehvNfXzAKBggqhkjOPQQDAgNJADBG\r\n"
+        "AiEAq7qkdvIiBRbuxt2OhCM8JAuxIFNizDWOwrhs1BvLcAsCIQCTqc9j3nfg7Fe4\r\n"
+        "tqL6SHP13wYzaHFkAcx8cpbIfxtU2g==\r\n"
+        "-----END CERTIFICATE-----\r\n",
+        "-----BEGIN EC PRIVATE KEY-----\r\n"
+        "Proc-Type: 4,ENCRYPTED\r\n"
+        "DEK-Info: AES-256-CBC,30E70C18B34A513D2B3D28B96E8C3ECB\r\n"
+        "\r\n"
+        "t2ywX2tbNw3fLMl2pa/lbf45RHecGQPDcVUCzrhro/JGq9xKpXLoWLwwa+sYrpVv\r\n"
+        "5W9tYSx3oCehrm8Pq44dv65JIb39Vy1OAhcsczbQRFZ6uh7mp+YH1zj3+juQhqRS\r\n"
+        "LjmOu0vJBPTnuEeN5hlXrPuymU9cez1RQBwu8vRoquY=\r\n"
+        "-----END EC PRIVATE KEY-----\r\n",
+        "eve"
+    },
+    {
+        { (const uint8_t*)"trudy", sizeof("trudy") - 1 },
+        "-----BEGIN CERTIFICATE-----\r\n"
+        "MIIBYzCCAQkCCQCzbzjgbS2cBzAKBggqhkjOPQQDAjBIMQswCQYDVQQGEwJVUzEV\r\n"
+        "MBMGA1UEBwwMRGVmYXVsdCBDaXR5MQwwCgYDVQQKDANEUFMxFDASBgNVBAMMC0RQ\r\n"
+        "UyBUZXN0IENBMB4XDTE4MDEzMTE4MDgzNloXDTI4MDEyOTE4MDgzNlowKzELMAkG\r\n"
+        "A1UEBhMCVVMxDDAKBgNVBAoMA0RQUzEOMAwGA1UEAwwFdHJ1ZHkwWTATBgcqhkjO\r\n"
+        "PQIBBggqhkjOPQMBBwNCAATxtnVcKFYOE8KuAT/GpTe7C4xcjPLM/HD9bZ62cFA7\r\n"
+        "eDrrLUV3GSh6ZQH0o8+bIisEXxElwD9AyCbfc5rTXLPaMAoGCCqGSM49BAMCA0gA\r\n"
+        "MEUCIQC97p99h23R686Vv37DSgcFrdKcMbneXnxrJQiSwCdL4wIgDEtGAoptwpqi\r\n"
+        "Cf7N//U4ttUfghfr+pQMyCDI0XkbGJk=\r\n"
+        "-----END CERTIFICATE-----\r\n",
+        "-----BEGIN EC PRIVATE KEY-----\r\n"
+        "Proc-Type: 4,ENCRYPTED\r\n"
+        "DEK-Info: AES-256-CBC,2F3AF19B8633A6F6DB8A4017A0ADB449\r\n"
+        "\r\n"
+        "cFtQdTFzfJqa5uXyoOEdIe6EjCN3J3/bt8b3gJPPC2BBehcL3Q5qZItfdnB+R/Qq\r\n"
+        "ROg8pmcA4LBzCVqK5AMaaK45sviqiOvdo+4oCIQMnLAQxATIvQKfmYmJImb8SIHq\r\n"
+        "CTU3rrbzthxlUF/JeFSI/8vf7n1dOTO8cZbIZAZegnE=\r\n"
+        "-----END EC PRIVATE KEY-----\r\n",
+        "trudy"
+    },
+    {
+        { NULL, 0 }, NULL, NULL, NULL
```

Comment:
checking that I understand the structure here correctly.
each Id consists of 
{ keyId, number of characters in the keyId}
x509 certificate,
Private Key for certificate,
Password (Just happens to be same as key ID for this test)

Can you verify that I got this correct.

Reply:
Correct